src/alert_tramsitter.py

```diff
@@ -0,0 +1,25 @@
+class AlertTransmitter():
+
+    def __init__(self, alertTarget):
+        self.alertTarget = alertTarget
+        self.alerter_function = self.set_alerter_function()
+
+    def set_alerter_function(self,):
+        alerter_function_name = f"alert_{self.alertTarget.lower()}"
+        alerter_function = getattr(AlertTransmitter, alerter_function_name)
+        return alerter_function
+
+    def send_alert(self, breachType):
+        self.alerter_function(breachType)
+
+    @staticmethod
+    def alert_to_email(breachType):
```

Comment:
how do you test and verify that an email has been sent?
Test cases might cover the paths but unable to verify!!
This can be improved...

Reply:
Thanks for the feedback, will improve this.

Reply:
Just a query, I already check if the proper alert function is set to "send_alert()" in one the tests. "send_alert()" can either be the function "alert_to_email" or "alert_to_controller". And the code always called send_alert(). 

Doesn't this mean we are already checking the correct alert function is being called?

Reply:
Setting right function and sending alert happens independent of each other(as i understand ). So to complete your verification, set and alert call both should be verified in one single test. This makes sure that whatever you set it what is getting executed.

Reply:
Hi @prithamvshetty 

I have made those changes, now it checks whether the proper function is called when you call send_alert.

regarding "alert_trasnmitter.py"
I could have used an if-else check to set the proper function or just made the user call the function explicitly, but both of them didn't seem like good approaches. am I right thinking this way or am I over complicating the code?